### PR TITLE
FAQ #3

### PR DESCRIPTION
**Pull request by jukent**

### CODE REVIEW

File: site/posts/python-tutorial-faq-part-3.ipynb [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    "\n",
+    "-----\n",
+    "<!-- TEASER_END -->"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "source": [
+    "**Q.** Are the spaces necessary around operators or  after commas?\n",
```
Comment (kmpaul) [inline site/posts/python-tutorial-faq-part-3.ipynb:26]:
```suggestion
    "**Q.** Are the spaces necessary around operators or after commas?\n",
```

File: site/posts/python-tutorial-faq-part-3.ipynb [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    "-----"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "source": [
+    "**Q.** What happens if you don’t explicitly specify a return value?\n",
+    "\n",
+    "**A.** If you don’t want to return anything you don’t need to have a return statement at the end of your function. By default, that means the function will return None.\n",
```
Comment (kmpaul) [inline site/posts/python-tutorial-faq-part-3.ipynb:39]:
```suggestion
    "**A.** If you don’t want to return anything you don’t need to have a `return` statement at the end of your function. By default, that means the function will return `None`.\n",
```

File: site/posts/python-tutorial-faq-part-3.ipynb [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    "**A.** Python uses garbage collection for memory management. That means that those variables will no longer be referenced anywhere, so they will be cleaned up whenever garbage collection happens which is not necessarily after the function return or during the execution of your script as long as there is no reason to clear memory. Essentially garbage collection is complicated, and controlling memory with Python is a hard thing to do.\n",
+    "\n",
+    "-----"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "source": [
+    "**Q.** In the `f-string` formatting, is `f` a function here?\n",
```
Comment (kmpaul) [inline site/posts/python-tutorial-faq-part-3.ipynb:83]:
```suggestion
    "**Q.** In the `f-string` formatting (i.e., `f'string'`), is `f` a function here?\n",
```

File: site/posts/python-tutorial-faq-part-3.ipynb [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "source": [
+    "**Q.** The `wc_data` and `wc_comp` are not defined. How does python know \n",
+    "what they refer to?\n",
+    "\n",
+    "**A.** They are defined in the for loop and are returned from the zip function when it is iterated over.\n",
```
Comment (kmpaul) [inline site/posts/python-tutorial-faq-part-3.ipynb:97]:
```suggestion
    "**A.** They are defined in the `for` loop and are returned from the `zip` function when it is iterated over.\n",
```

**Review by kmpaul — CHANGES_REQUESTED** [review]
These are just minor suggestions, mostly formatting for clarity.  It looks great, @jukent!

**Comment by jukent** [comment]
> These are just minor suggestions, mostly formatting for clarity. It looks great, @jukent!

Thanks for the edits!

**Comment by halehawk** [comment]
Do you need to mention in Python 3, the first question will get 2/2=1.0, but in Python 2, 2/2=1
In f string question, you have 2 would, please remove one.

**Comment by kmpaul** [comment]
That's a good point @halehawk. Maybe a mention that Python 2 is different, but that we don't recommend using Python 2 because it is deprecated.

File: site/posts/python-tutorial-faq-part-3.ipynb [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+    "Here is a compilation of questions from the third session of the Python Tutorial Seminar Series which covered writing functions as well as f-string formatting.\n",
+    "\n",
+    "-----"
+   ]
+  },
+  {
+   "source": [
+    "**Q.** Seeing how `2/2` returns `1.0` instead of `1`, is the numeric type automatically determined from the inputs or do you have to force it?\n",
+    "\n",
+    "**A.** The numeric type is inferred. `/` always returns a float for numeric types. `//` always returns an integer.\n",
```
Comment (kmpaul) [inline site/posts/python-tutorial-faq-part-3.ipynb:16]:
```suggestion
    "**A.** The numeric type is inferred. `/` always returns a float for numeric types. `//` always returns an integer. 
 One should note that Python 2 does not behave this way.  In Python 2, the `//` operator does not exist and the `/` operator behaves like it does in other languages (i.e., division of two integers yields an integer and division with a float yields a float).\n",
```

**Review by kmpaul — CHANGES_REQUESTED** [review]
Here's a first draft of some language regarding Python 2.

**Comment by jukent** [comment]
> Here's a first draft of some language regarding Python 2.

I think this answers the Python 2 nuance thoroughly enough for our audience.

**Review by kmpaul — APPROVED** [review]

**Pull request merged**